src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -426,6 +426,10 @@ private function validateAndDenormalize(string $currentClass, string $attribute,
                     return null;
                 }
 
+                if ('' === $data && Type::BUILTIN_TYPE_ARRAY === $type->getBuiltinType()) {
```

Comment:
```
                if ('' === $data) {
                 if ($type->isNullable() && \in_array($type->getBuiltinType(), [Type::BUILTIN_TYPE_BOOL, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT], true)) {
                    return null;
                  }

                  if (Type::BUILTIN_TYPE_ARRAY === $type->getBuiltinType()) {
                      return [];
                  }
                }
```

To prevent an extra check.

Reply:
Good idea, it really make sense! Thank you for the suggestion, Kévin. :+1: